src/main/frontend/fs.cljs

```diff
@@ -123,19 +125,51 @@
     :else
     (js/window.workerThread.rimraf dir)))
 
+(defonce secret "IRwamok0gbumjx41O0z83V/nzcqrac5vML6P62zS23c=")
+
+(defn new-nonce
+  []
+  (nacl/randomBytes nacl/secretbox.nonceLength))
+
+(defn encrypt
+  [key content]
+  (when key
+    (let [key-decoded (nacl-util/decodeBase64 key)
+          nonce (new-nonce)
+          content-decoded (nacl-util/decodeUTF8 content)
+          box (nacl/secretbox content-decoded nonce key-decoded)
+          full-message (new js/Uint8Array (+ nonce.length box.length))]
+      (js/console.log "nonce" (nacl-util/encodeBase64 nonce))
+      (js/console.log "box" (nacl-util/encodeBase64 box))
+      (.set full-message nonce)
+      (.set full-message box nonce.length)
+      (nacl-util/encodeBase64 full-message))))
+
+(defn decrypt
+  [key content-with-nonce]
+  (when key
+    (let [key-decoded (nacl-util/decodeBase64 key)
+          content-with-nonce-decoded (nacl-util/decodeBase64 content-with-nonce)
+          nonce (.slice content-with-nonce-decoded 0 nacl/secretbox.nonceLength)
+          content (.slice content-with-nonce-decoded nacl/secretbox.nonceLength content-with-nonce.length)
+          decrypted (nacl-util/encodeUTF8 (nacl/secretbox.open content nonce key-decoded))]
+      (println "decrypted" decrypted)
+      decrypted)))
+
 (defn read-file
   ([dir path]
    (read-file dir path (clj->js {:encoding "utf8"})))
   ([dir path option]
-   (cond
-     (local-db? dir)
-     (let [handle-path (str "handle" dir "/" path)]
-       (p/let [handle (idb/get-item handle-path)
-               local-file (and handle (.getFile handle))]
-         (and local-file (.text local-file))))
+   (decrypt secret
```

Comment:
The `cond` body returns a promise, we might need to resolve it first to get the encrypted text. Something like:
```clojure
(defn read-file
  ([dir path]
   (read-file dir path (clj->js {:encoding "utf8"})))
  ([dir path option]
   (p/let [encrypted (cond
                       (local-db? dir)
                       (let [handle-path (str "handle" dir "/" path)]
                         (p/let [handle (idb/get-item handle-path)
                                 local-file (and handle (.getFile handle))]
                           (and local-file (.text local-file))))

                       :else
                       (js/window.pfs.readFile (str dir "/" path) option))]
     (decrypt secret encrypted))))
```

Reply:
I see! Is this method used for reading from native file system too? Asking because I never see this method being invoked.
I'll try with a github sync now.

Reply:
It's used for both native file system and lightning-fs (work with indexeddb), you can "re-index" the repo which will output the following error because the passed parameter is a promise instead of a text :)
![Screenshot-2021-01-07_21:27:21](https://user-images.githubusercontent.com/479169/103898096-641f0780-512f-11eb-915e-57d8586842d4.png)

Reply:
Hmm, this is not what I see. When I "re-index" the repo, I always get this

![Screenshot from 2021-01-07 22-43-08](https://user-images.githubusercontent.com/17571/103899706-1f4c9e00-513a-11eb-8b12-c88df7c1ea6e.png)

because the config is not decrypted. I feel like my files are readed from somewhere else, but I can't track it down 😬

Reply:
Yeah, it seems to be the config is not decrypted. The reason is that the config is from the Datascript files db. There're two kinds of databases in logseq:
1. files db with the schema `frontend.db-schema/files-db-schema`
2. pages and blocks db with the schema `frontend.db-schema/schema`
https://github.com/logseq/logseq/blob/enhance/undo-redo/src/main/frontend/db_schema.cljs#L15
The reason to separate the databases is for better performance.

It seems that the config file in the file database is not decrypted yet.

Reply:
I can reproduce the parsing config error with a local folder, will investigate it soon.

Reply:
I think the problem is that the local files are not decrypted. 
Check both https://github.com/logseq/logseq/blob/master/src/main/frontend/handler/web/nfs.cljs#L126 and https://github.com/logseq/logseq/blob/master/src/main/frontend/handler/web/nfs.cljs#L242.

Reply:
Thanks! I check both before but couldn't make it work. I think I know why now, because they are also promises! I'll test right now.

Reply:
Cool! Something like this should work:
``` clojure
(p/let [content (.text (:file/file file))
          content (fs/decrypt fs/secret content)]
    (assoc file :file/content (fs/decrypt fs/secret content)))
```

Reply:
This might help, the documentation of the `promesa` library we're using: https://funcool.github.io/promesa/latest/user-guide.html :smiley:

Reply:
Great! This is enough for me to make some progress this weekend 😄

Reply:
Awesome!